### PR TITLE
Add CBMC proof of inSelfTestHandler

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of inSelfTestHandler function

Description
-----------
This PR adds files required for the CBMC proof of inSelfTestHandler function

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/source/stubs.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    return err;
+}
+
+OtaOsStatus_t startTimerStub( OtaTimerId_t otaTimerId,
+                              const char * const pTimerName,
+                              const uint32_t timeout,
+                              OtaTimerCallback_t callback )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:48]: I think status should be initailized here.

File: test/cbmc/source/stubs.c [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    __CPROVER_assert( pTimerName != NULL, "Error: TimerName cannot be NULL" );
+
+    __CPROVER_assert( callback != NULL, "Error: callback should be pointing to a function." );
+
+    return status;
+}
+
+OtaOsStatus_t stopTimerStub( OtaTimerId_t otaTimerId )
+{
+    OtaOsStatus_t status;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:62]: I think status should be initailized here.

File: test/cbmc/source/stubs.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    __CPROVER_assert( ( otaTimerId != OtaSelfTestTimer ) || ( otaTimerId != OtaRequestTimer ),
+                      "Error: Expected otaTimerId to be either OtaSelfTestTimer or OtaRequestTimer." );
+
+    return status;
+}
+
+OtaPalImageState_t getPlatformImageStateStub( OtaFileContext_t * const pFileContext )
+{
+    OtaPalImageState_t state;
```
Comment (gshvang) [inline test/cbmc/source/stubs.c:72]: State should be initailized here.

**Review by gshvang — APPROVED** [review]

**Review by gshvang — APPROVED** [review]

File: test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    pEventData = ( OtaEventData_t * ) malloc( sizeof( OtaEventData_t ) );
+
+    otaInterface.os.timer.stop = stopTimerStub;
+    otaInterface.pal.getPlatformImageState = getPlatformImageStateStub;
+    otaInterface.pal.reset = resetPalStub;
+
+    /* OtaInterfaces and the interfaces included in it cannot be NULL and they are checked
+     * during the initialization of OTA specifically in the OTA_Init function. */
+    otaAgent.pOtaInterface = &otaInterface;
```
Comment (paulbartell) [inline test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c:49]: The remainder of the otaAgent struct should be unconstrained.

File: test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    /* Initialize OtaAppCallback to an empty callback function. */
+    otaAgent.OtaAppCallback = otaAppCallbackStub;
+
+    err = inSelfTestHandler( pEventData );
+
+    /* inSelfTestHandler returns the values which follow OtaErr_t enum. If it does not, then
+     * there is a problem. */
+    __CPROVER_assert( ( err >= OtaErrNone ) && ( err <= OtaErrActivateFailed ),
+                      "Invalid return value from inSelfTestHandler: Expected a value from OtaErr_t enum." );
+
```
Comment (paulbartell) [inline test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c:60]: The otaAgent.fileContext.isInSelfTest flag is not checked.

File: test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    free( pEventData );
```
Comment (paulbartell) [inline test/cbmc/proofs/inSelfTestHandler/inSelfTestHandler_harness.c:61]: The resulting imageState and reason are not validated

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**